src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -1397,7 +1397,7 @@ public struct SyntaxNodeAnalysisContext
         /// <summary>
         /// <see cref="CodeAnalysis.Compilation"/> containing the <see cref="SyntaxNode"/>.
         /// </summary>
-        public Compilation? Compilation => _semanticModel?.Compilation;
+        public Compilation Compilation => _semanticModel?.Compilation ?? throw new InvalidOperationException();
```

Comment:
Can `_semanticModel` ever be null? Can we check and throw an `ArgumentNullException` in the `SyntaxNodeAnalysisContext` constructors below?

Reply:
`_semanticModel` is already annotated to not be null from construction. However, this is a value type so the `?.` here is used to avoid a `NullReferenceException` since non-defaultable value types aren't a thing yet.

Reply:
Could you clarify why it is more desirable to null-check and throw InvalidOperationException versus not checking and just letting the NullReferenceException happen?

Reply:
It could really go either way. Eventually I'd like to see non-defaultable value types cover the scenario.